model/build-processor/src/main/java/org/keycloak/models/map/processor/AbstractGenerateEntityImplementationsProcessor.java

```diff
@@ -123,11 +123,15 @@ private void processTypeElement(TypeElement e) {
 
         // Merge plurals with singulars
         methodsPerAttribute.keySet().stream()
-                .filter(key -> methodsPerAttribute.containsKey(key + "s"))
+                .filter(key -> methodsPerAttribute.containsKey(key + "s") || (key.endsWith("y") && methodsPerAttribute.containsKey(key.substring(0, key.length() -1) + "ies")))
```

Comment:
Could the filter to check whether the property is plural and the transformer from plural to singular be extracted into two methods, `hasPlural` and `pluralToSinglular`, perhaps in `Util` class? They would be used in `FieldAccessorType` as well.